src/password/tests.py

```diff
@@ -0,0 +1,335 @@
+        client = Client()
+        username = os.environ.get('LOGIN_USERNAME')
+        old_password = os.environ.get('LOGIN_PASSWORD')
+        new_password = os.environ.get('LOGIN_PASSWORD')
+        body = {
+            'auth_token': os.environ.get('LOGIN_SEC_TOKEN'),
+            'old': os.environ.get('LOGIN_PASSWORD'),
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch('cassy.change_user_password')
+    def test_password_change_with_mail_token_invalid_same_old_and_new(self, cassy_mock):
+        """
+        Tests the case where a user logs in after having gotten a password reset email.
+        """
+        setup_test_environment()
+        client = Client()
+        username = os.environ.get('LOGIN_USERNAME')
+        old_password = os.environ.get('LOGIN_PASSWORD')
+        new_password = os.environ.get('LOGIN_PASSWORD')
+        body = {
+            'token': os.environ.get('LOGIN_SEC_TOKEN'),
+            'username': username,
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch('cassy.change_user_password')
+    @patch('cassy.verify_auth_token')
+    def test_password_change_with_admin_invalid_same_old_and_new(self, cassy_auth, cassy_mock):
+        """
+        Tests the case where an admin resets someone's password.
+        Mocking the admin authentication.
+        """
+        setup_test_environment()
+        client = Client()
+        cassy_auth.return_value = 'admin'
+        username = 'mr.forgetful'
+        old_password = 'testme'
+        new_password = 'testme'
+        auth_token = 'token'
+        body = {
+            'auth_token': auth_token,
+            'username': username,
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        cassy_auth.assert_called_once_with(auth_token)
+        self.assertEqual(response.status_code, 403)
+
+    def test_password_reset_valid_username(self):
+        """
+        Tests the password reset endpoint.
+        """
+        setup_test_environment()
+        client = Client()
+        body = {
+            'username': 'welches',
+        }
+        reset_response = client.post(
+            '/password/reset',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        response_body = json.loads(reset_response.content.decode('utf-8'))
+        reset_token = response_body.get('reset_token', '')
+        final_response = client.get(
```

Comment:
If we don't return the token in the response (my recommendation), we can still test this with a mock and assert send_mail() was called with what we expect.